### PR TITLE
Improve input validation

### PR DESCRIPTION
**Pull request by mbien**
Add early input validation for weblog handles and similar parameters so that roller can respond early instead of responding later in the business logic. 

Output encode client provided strings.

Improved Exception handling / Error page mapping.

### CODE REVIEW

**Comment by mbien** [comment]
force pushed

* switched from owasp encoder to commons StringEscapeUtils.escape* - no need to add another library for this simple usecase
* validation is currently fairly strict (too strict?) and enforces alphanumeric Strings for blog handles, params and similar tokens

File: app/src/main/java/org/apache/roller/weblogger/webservices/tagdata/TagDataServlet.java [hunk 1]
```diff
@@ -70,36 +72,56 @@ public void doGet(
...
+        }
+        
+        String format = "json";  // default
+        
         if (request.getParameter("format") != null) {
+            
             format = request.getParameter("format");
+            if(!format.equals("json") || !format.equals("xml")) {
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Malformed URL");
+                return;
```
Comment (mbien) [inline app/src/main/java/org/apache/roller/weblogger/webservices/tagdata/TagDataServlet.java:112]: todo: this should be && not ||

**Comment by snoopdave** [comment]
Lot's of good changes here. I went through commit-by-commit and did not see any problems.

**Review by snoopdave — APPROVED** [review]
LGTM

**Comment by mbien** [comment]
thank you for the review @snoopdave

**Pull request merged**